lib/elixir/lib/task.ex

```diff
@@ -183,24 +183,40 @@ defmodule Task do
     * `:ref` - the task monitor reference
 
     * `:owner` - the PID of the process that started the task
-
   """
+  @enforce_keys [:pid, :ref, :owner]
   defstruct pid: nil, ref: nil, owner: nil
 
-  @type t :: %__MODULE__{}
+  @typedoc """
+  The Task type.
+
+  See `%Task{}` for information about each field of the structure.
+  """
+  @type t :: %__MODULE__{
+          pid: pid() | nil,
+          ref: reference() | nil,
+          owner: pid() | nil
+        }
 
   @doc """
   Returns a specification to start a task under a supervisor.
 
-  See `Supervisor`.
+  `arg` is passed as the argument to `Task.start_link/1` in the `:start` field
+  of the spec.
+
+  For more information, see the `Supervisor` module,
+  the `Supervisor.child_spec/2` function and the `t:Supervisor.child_spec/0` type.
   """
   @doc since: "1.5.0"
+  @spec child_spec(term) :: Supervisor.child_spec()
   def child_spec(arg) do
-    %{
+    spec = %{
       id: Task,
       start: {Task, :start_link, [arg]},
       restart: :temporary
     }
+
+    Supervisor.child_spec(spec, [])
```

Comment:
Why this change?

Reply:
I think it is trying to be consistent with other modules but other modules do not accept options here, so this change is not necessary.

Reply:
sorry, I thought a map with default values was created by Supervisor.child_spec/2, that's why I passed it through